src/segment/filter/blocked_bloom/mod.rs

```diff
@@ -128,6 +128,30 @@ impl<'a> BlockedBloomFilterReader<'a> {
     pub fn contains(&self, key: &[u8]) -> bool {
         self.contains_hash(Self::get_hash(key))
     }
+
+    /// Returns `true` if any prefix of the key may be contained.
+    ///
+    /// Returns `None` if the key is out of domain.
+    #[must_use]
+    #[inline]
```

Comment:
I generally don't put inline statements because LLVM is pretty aggressive about it anyway except for when I can actually prove that it is not inlined normally (e.g. in godbolt), so I would just remove those.

Reply:
removed

Reply:
need to fix this, this was re-added with the revert

Reply:
fixed for real now